RELEASE.md

```diff
@@ -0,0 +1,55 @@
+# OpenTelemetry Release Process
+
+This repository uses semantic versioning. Please keep this in mind when choosing version numbers.
+
+1. **Alert others you are releasing**
+
+   There should be no commits made to master while the release is in progress (about 10 minutes). Before you start
+   a release, alert others on [gitter](https://gitter.im/open-telemetry/opentelemetry-java) so that they don't accidentally
+   merge anything. If they do, and the build fails because of that, you'll have to recreate the release tag described below.
+
+1. **Update CHANGELOG**
+
+   Update CHANGELOG.md with a list changes since the last release. Each entry must include the release number,
```

Comment:
NIt: add an example of one entry to keep it consistent.

Reply:
Good point, will add (and we also need to add an actual CHANGELOG)